src/validator.rs

```diff
@@ -5,7 +5,7 @@ use crate::{
 use serde_json::{Map, Value};
 use std::fmt;
 
-pub trait Validate: Send + Sync + ToString {
+pub trait Validate<'context>: Send + Sync + ToString {
```

Comment:
> NOTE: In order to achieve it by ensuring that we're limiting clone (for example in $ref keyword) we need to have a specific lifetime into Validate trait, but overall I feel like it was an acceptable tradeoff to have.

Could you, please, elaborate on why exactly do we need this lifetime? Not sure if I completely understand what clones you mean

Reply:
Currently `RefValidator` builds the `validators` of the reference only when invoked for validation (and we have the `RWLock` to ensure that eventually we would compile it only once) ([here](https://github.com/Stranger6667/jsonschema-rs/blob/master/src/keywords/ref_.rs#L39-L40)).
This means that we need to preserve the config information within the `RefValidator` instance.

Let's assume that we don't have a lifetime, then we need to clone it, instead by enforcing the lifetime we can just store it via a CoW instance.

We might get rid of it if we can pull the build config from the JSONSchema passed along during validation, but I did not attempt it as I was looking for a general feedback on the idea first :)

Definitely if we can remove the lifetimes and ensuring that we're not cloning info then it would be preferrable

Reply:
Cross-checked again and I don't know why I kept it ... we actually don't need it (at least not for now).

So I'm going to remove it and keep this PR scoped